packages/core/src/amazonqFeatureDev/controllers/chat/controller.ts

```diff
@@ -306,19 +293,78 @@ export class FeatureDevController {
                         },
                     ],
                 })
-            } else {
-                const errorMessage = createUserFacingErrorMessage(
-                    `${featureName} request failed: ${err.cause?.message ?? err.message}`
-                )
+                break
+            default:
+                switch (session?.state.phase) {
+                    case DevPhase.APPROACH:
+                        if (isDenyListedError) {
+                            defaultMessage = ErrorMessages.approachPhase.denyListedError
+                        } else {
+                            defaultMessage = ErrorMessages.approachPhase.default
+                        }
+                        break
+                    case DevPhase.CODEGEN:
+                        if (isDenyListedError) {
+                            defaultMessage = ErrorMessages.codeGen.denyListedError
+                            noRetries = true
+                        } else {
+                            defaultMessage = ErrorMessages.codeGen.default
+                        }
+                        break
+                }
+
                 this.messenger.sendErrorMessage(
-                    errorMessage,
+                    defaultMessage ? defaultMessage : errorMessage,
                     message.tabID,
-                    this.retriesRemaining(session),
-                    session?.state.phase,
-                    session?.conversationIdUnsafe
+                    noRetries ? 0 : this.retriesRemaining(session),
+                    session?.conversationIdUnsafe,
+                    !!defaultMessage
```

Comment:
Should this not be `session?.conversationIdUnsafe` being passed here?

Reply:
public sendErrorMessage in messenger have the following attributes:
```
errorMessage: string,
tabID: string,
retries: number,
conversationId?: string,
showDefaultMessage?: boolean
```
session?.conversationIdUnsafe is being passed as the conversationId, which match the practice being used in other places. 

Ex: 
```
private async generateCodeClicked(message: any) {...
this.messenger.sendErrorMessage(
                errorMessage,
                message.tabID,
                this.retriesRemaining(session),
                session?.conversationIdUnsafe
            )
```